File: design-pattern-go/src/patterns/creational/abstract_factory/FordFactory.go
package abstract_factory

type FordFactory struct {
	VehicleFactory
}

func (f *FordFactory) CreateDeportiveCar() DeportiveCar {
	return FordMustang{}
}
func (f *FordFactory) CreateRusticCar() RusticCar {
	return FordPickUp{}
}

File: design-pattern-ts/src/patterns/creator/abstract-factory/FordFactory.ts
import { DeportCar } from "./DeportCar"
import { RusticCar } from "./RusticCar"
import { VehicleFactory } from "./VehicleFactory"
import { FordMustang } from "./FordMustang"
import { FordExplorer } from "./FordExplorer"

export class FordFactory implements VehicleFactory {
	produceDeportCar(): DeportCar {
		return new FordMustang()
	}
	produceRusticCar(): RusticCar {
		return new FordExplorer()
	}
}
